draft-ietf-mls-protocol.md

```diff
@@ -1867,6 +2018,43 @@ A member of the group applies a Remove message by taking the following steps:
 
 * Blank the intermediate nodes along the path from the removed leaf to the root
 
+### PSK
+
+A PSK proposal can be used to request that a PSK be injected into the
+key schedule in the process of advancing the epoch.
+
+~~~~~
+struct {
+    PreSharedKeyID psk;
+} PSK;
+~~~~~
+
+When processing a commit message including one or more PSK proposals, a group
+member MUST derive the `psk_secret` for the inclusion in the Key Schedule as
+described in {{pre-shared-keys}}, where the order of the PSKs corresponds to the
+order of the PSK proposals in the commit.
+
+### ReInit
+
+A ReInit proposal represents a request to re-initialize the group with different
+parameters, for example, to increase the version number or to change the
+ciphersuite. The re-initialization is done by creating a completely new group
+and shutting down the old one.
+
+~~~~~
+struct {
+    opaque group_id<0..255>;
+    ProtocolVersion version;
+    CipherSuite cipher_suite;
+    Extension extensions<0..2^32-1>;
+} ReInit;
+~~~~~
+
+A member of the group applies a ReInit message by waiting for the committer to
+send the Welcome message and by checking that the `group_id` and the parameters
+of the new group corresponds to the ones specified in the proposal.
```

Comment:
Do we want to specify a PSK that carries over from the old group to the new group?  So you would define here something like: The new Welcome MUST have a PreSharedKey extension with a `reinit` entry pointing to the epoch initiated by the Commit that covers the ReInit proposal.

Reply:
That's exactly what we want to do, so you're right indeed.